### PR TITLE
TPC: Trigger handling for non-interleaved data with dense format

### PR DESCRIPTION
**Pull request by wiechula**

### CODE REVIEW

**Comment by wiechula** [comment]
@davidrohr , are these changes fine for you?

**Comment by wiechula** [comment]
The failure looks unrelated.
@davidrohr , can you have a look if the trigger handling is fine like this?

File: GPU/GPUTracking/Base/GPUReconstructionConvert.cxx [hunk 1]
```diff
@@ -695,7 +695,11 @@ void zsEncoderImprovedLinkBased::decodePage(std::vector<o2::tpc::Digit>& outputB
     if (!tbHdr->isLinkZS()) {
       throw std::runtime_error("ZS TB Hdr does not have linkZS magic word");
     }
-    int timeBin = (decHDR->timeOffset + tbHdr->bunchCrossing + (unsigned long)(o2::raw::RDHUtils::getHeartBeatOrbit(*rdh) - firstOrbit) * o2::constants::lhc::LHCMaxBunches) / LHCBCPERTIMEBIN;
+    int timeBin = (int(decHDR->timeOffset) + int(tbHdr->bunchCrossing) + (int)(o2::raw::RDHUtils::getHeartBeatOrbit(*rdh) - firstOrbit) * o2::constants::lhc::LHCMaxBunches - triggerBC) / LHCBCPERTIMEBIN;
+    if (timeBin < 0) {
+      LOGP(info, "zsEncoderImprovedLinkBased::decodePage skipping digits hdr->tOff {} + hdr->bc {} + (orbit {} - firstOrbit {}) * maxBunch {} - triggerBC {} = {} < 0", decHDR->timeOffset, tbHdr->bunchCrossing, o2::raw::RDHUtils::getHeartBeatOrbit(*rdh), firstOrbit, o2::constants::lhc::LHCMaxBunches, triggerBC, timeBin);
```
Comment (davidrohr) [inline GPU/GPUTracking/Base/GPUReconstructionConvert.cxx]: Can this create an info message per raw page? If so, we need to downscale it or reduce it to debug.

File: GPU/GPUTracking/Base/GPUReconstructionConvert.cxx [hunk 2]
```diff
@@ -953,10 +957,14 @@ void zsEncoderDenseLinkBased::decodePage(std::vector<o2::tpc::Digit>& outputBuff
     int encodeBitsBlock = v2Flag ? v2nbits : encodeBits;
     decPagePtr += (nTotalSamples * encodeBitsBlock + 7) / 8;
 
-    if (linkBC < triggerBC || nTotalSamples == 0) {
+    // time bin might be smaller 0 due to triggerBC
+    int timeBin = (int(linkBC) + (int)(o2::raw::RDHUtils::getHeartBeatOrbit(*rdh) - firstOrbit) * o2::constants::lhc::LHCMaxBunches - int(triggerBC)) / LHCBCPERTIMEBIN;
+    if (timeBin < 0 || nTotalSamples == 0) {
+      if (timeBin < 0 && nTotalSamples > 0) {
+        LOGP(info, "zsEncoderDenseLinkBased::decodePage skipping digits (linkBC {} + orbit {} - firstOrbit {}) * maxBunch {} - triggerBC {} = {} < 0, nTotalSamples {}", linkBC, o2::raw::RDHUtils::getHeartBeatOrbit(*rdh), firstOrbit, o2::constants::lhc::LHCMaxBunches, triggerBC, timeBin, nTotalSamples);
```
Comment (davidrohr) [inline GPU/GPUTracking/Base/GPUReconstructionConvert.cxx]: same as above

**Comment by wiechula** [comment]
In principle I can demote both as debug.

**Comment by wiechula** [comment]
@davidrohr , the error looks unrelated, is it fine like this, or do you have further comments?

**Pull request merged**